extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java

```diff
@@ -255,6 +260,17 @@ private Response respond(List<ResourceDescription> descriptions) {
         return Response.status(Status.NOT_FOUND).build();
     }
 
+    private String adjustRoot(String basePath) {
+        if (httpRoot.equals("/")) {
+            //leave it alone
+            return basePath;
+        }
+        if (basePath.startsWith("/")) {
+            return httpRoot + basePath;
+        }
+        return httpRoot.substring(1) + "/" + basePath;
```

Comment:
I might be missing something obvious but why the substring in this very particular case?

Reply:
It's not obvious, for this case the template already has a '/' as it is expecting a non-slash prefixed path. As the httpRoot starts with '/' we need to remove it.